
## Pipeline introduction

<!-- The single-case report that you are reading  -->
<!-- contains data for one patient  -->
<!-- fro the  -->
<!-- \colorbox{kispiblue!30}{\texttt{traditional clinical genetics}} module shown in figure \ref{fig:qv}. -->
<!-- This modules is just one of several stages within a larger study figure \ref{fig:qv}. -->
<!-- The larger study consists of thorough multi-part analysis to reach the final combination of (1) newly identified cohort-level genes associated with disease with (2) single case-level known disease-causing results. -->
<!-- Additionally, several QV protocols are piped together to create increasingly filtered datasets to match the needs at a certain stage of analysis.  -->
<!-- Different analyses are merged cover multiple scenarios.  -->
<!-- In the complete study, we have planned a complex analysis pipeline using -->
<!-- \colorbox{kispiblue!30}{\texttt{QV SNV/INDEL}} +  -->
<!--   \colorbox{kispiblue!30}{\texttt{QV CNV}} +  -->
<!--   \colorbox{kispiblue!30}{\texttt{QV structural variation}} +  -->
<!--   \colorbox{kispiblue!30}{\texttt{QV rare disease known}} +  -->
<!--   \colorbox{kispiblue!30}{\texttt{QV statistical association QC}},  -->
<!--   merged for a thorough multi-part analysis to reach the final combination of (1) newly identified cohort-level genes associated with disease with (2) single case-level known disease-causing results. -->
This protocol is featured in the pre-prints
[^lawless2025qv],
[^lawless2025genomicdesign], [^lawless2025genomicbenchmark].
The single-case report that you are currently reading 
comes from the SNV/INDEL pipeline \colorbox{colorSUNSET1!30}{\texttt{pipeline DNA SNV INDEL v1}}. 
It focuses on data from one patient from the end-point module
\colorbox{kispiblue!30}{\texttt{traditional clinical genetics}}, as illustrated in Figure 
\ref{fig:swisspedhealth_gemomics_plan} 
(generalised in figures \ref{fig:varanno} - \ref{fig:hgvs}). 
This module is one part of a comprehensive study that encompasses several stages, each contributing uniquely to the overall analysis.
This broader study involves a detailed, multi-part analysis aimed at achieving two primary objectives: (1) identifying new cohort-level genes associated with disease and (2) confirming known disease-causing results at the single-case level.
To facilitate this, we employ a series of Quality Verification (QV) protocols, designed to progressively refine datasets to meet specific analytical requirements at various stages of the study. These analyses integrate multiple scenarios, ensuring a robust examination of the data.
Specifically, the complete study includes the following components, merged to support a thorough investigation:
\colorbox{kispiblue!30}{\texttt{QV SNV/INDEL}} +
\colorbox{kispiblue!30}{\texttt{QV CNV}} +
\colorbox{kispiblue!30}{\texttt{QV Structural Variation}} +
\colorbox{kispiblue!30}{\texttt{QV Rare Disease Known}} +
\colorbox{kispiblue!30}{\texttt{QV Statistical Association QC}}.
Each of these components plays a critical role in our multi-faceted analysis strategy, aimed at uncovering novel genetic associations and validating known markers of disease.

\begin{figure}[h]
\centering
\includegraphics[width=\textwidth]{./images/misc/qv_pipeline_vcurrent.pdf}
\caption{Summary of the design DNA SNV INDEL v1 pipeline. QV1 and QV2 are shown as sequential and potentially piped protocol steps.}
\label{fig:pipeline_figure}
\end{figure}

## Qualifying variants protocol

This analysis focused on a SNV/INDEL pipeline 
\colorbox{colorSUNSET1!30}{\texttt{pipeline DNA SNV INDEL v1}}
using two QV sets named 
\colorbox{colorSUNSET2!60}{\texttt{QV SNV INDEL v1}} and
\colorbox{colorSUNSET2!60}{\texttt{QV SNV INDEL v2}} as shown in
figure \ref{fig:qv}.

A cohort of patient WGS data was analysed to identify genetic determinants for the clinical diagnosis of multiple phenotypes. 
This pipeline is concerned with WGS germline short variant discovery (SNVs + Indels) and interpretation.
\underline{First, a flexible QV set (v1)} was used for cohort-level statistical genomics and \underline{second a rare disease QV set (v2)} was used for single-case analysis.
(\textbf{1}) Data was processed with the 
\colorbox{colorSUNSET1!30}{\texttt{pipeline DNA SNV INDEL v1}} pipeline, which implements
(\textbf{a}) \colorbox{colorSUNSET2!60}{\texttt{QV SNV INDEL v1}} criteria, resulting in the prepared dataset
\colorbox{colorSUNSET3!30}{\texttt{dataset DNA SNV INDEL v1}}.
(\textbf{b}) The dataset was subsequently analysed in combination with other modules including 
\colorbox{colorSUNSET4!30}{\texttt{PCA SNV INDEL v1}} and 
\colorbox{colorSUNSET5!30}{\texttt{statistical genomics v1}} to complete statistical analysis on a joint cohort.
(\textbf{2}) Next, 
the prepared dataset (from step 1a)
\colorbox{colorSUNSET3!30}{\texttt{Dataset DNA SNV INDEL v1}} was processed further with more strict filtering using
\colorbox{colorSUNSET2!60}{\texttt{QV SNV INDEL v2}} to identify previously known causal genetic variants for each patient based on disease-gene panel and curated evidence sources, resulting in \colorbox{colorSUNSET3!30}{\texttt{Dataset DNA SNV INDEL v2}} and final interpretation in
\colorbox{colorSUNSET5!30}{\texttt{single case report SNV INDEL v1}}.

\begin{tcolorbox}[
  % breakable,  % Allows the box to split over pages
  colback=white!0,  % No background color (fully transparent)
  colframe=black,  % Black border color
  boxrule=1pt,  % Width of the border
  arc=1mm,  % Radius of the corner rounding
  outer arc=1mm,
  title=\textbf{Diagrammatic representation}
]

\dirtree{%
  .1 \colorbox{colorSUNSET1!30}{\texttt{pipeline DNA SNV INDEL v1}}.
  .2 Flexible QV criteria.
  .3 \colorbox{colorSUNSET2!60}{\texttt{QV SNV INDEL v1}} (-> \colorbox{colorSUNSET3!30}{\texttt{dataset DNA SNV INDEL v1}}).
  .4 \colorbox{colorSUNSET4!30}{\texttt{PCA SNV INDEL v1}}.
  .4 \colorbox{colorSUNSET5!30}{\texttt{statistical genomics v1}} -> Result 1.
  .3 \colorbox{colorSUNSET3!30}{\texttt{dataset DNA SNV INDEL v1}}.
  .4 Rare disease QV criteria.
  .5 \colorbox{colorSUNSET2!60}{\texttt{QV SNV INDEL v2}} (-> \colorbox{colorSUNSET3!30}{\texttt{dataset DNA SNV INDEL v2}}).
  .6 \colorbox{colorSUNSET5!30}{\texttt{single case report SNV INDEL v1}} -> Result 2.
}

Result 1 = Association signal (variant P-value). \\
Result 2 = Single variant for single patient.
\end{tcolorbox}


The pipeline is illustrated in figure \ref{fig:pipeline_figure}.
The study data flow is shown in 
figure \ref{fig:swisspedhealth_gemomics_plan}.

\begin{figure}[h]
\centering
  \includegraphics[width=0.8\textwidth]{./images/misc/swisspedhealth_gemomics_plan.pdf}
\caption{Study data flow.}
\label{fig:swisspedhealth_gemomics_plan}
\end{figure}

<!-- % \begin{figure}[h] -->
<!-- % \centering -->
<!-- % \includegraphics[width=\textwidth]{path/to/variant-filtering-workflow.png} -->
<!-- % \caption{Illustration of QV workflow .This figure summarizes the variant filtering and annotation processes, providing a snapshot of total qualifying variant and gene counts by ethnicity and for overall cases and controls.} -->
<!-- % \end{figure} -->

<!-- % \begin{figure}[h] -->
<!-- % \centering -->
<!-- %  \includegraphics[width=\textwidth]{./images/qv_filter_pyramid_vcurrent.pdf} -->
<!-- % \caption{qv filter pyramid vcurrent} -->
<!-- % \label{fig:qv_filter_pyramid_vcurrent} -->
<!-- % \end{figure} -->







[^lawless2025qv]: Lawless, et al., 2025. (QV protocol) Conceptualising qualifying variants for genomic analysis. Unpublished. DOI: []().
[^lawless2025genomicdesign]: Lawless, et al., 2025. (Genomics design) Pediatric precision medicine with SwissPedHealth’s genomics programme. DOI: []().
[^lawless2025genomicbenchmark]: Lawless, et al., 2025. (Genomics benchmark result) Advancing pediatric precision medicine with SwissPedHealth’s integrated databases and automated genomic pipelines. Unpublished. DOI: []().
